update_sample_status/main.py

```diff
@@ -43,26 +51,60 @@ def update_airtable(project_config: dict, sample: str, status: str):
     return ('', 204)
 
 
-def update_sm(project: str, sample: str, status: str, batch: str):
-    """Update the status of a sequence given it's corresponding
-    external sample id"""
+def update_sm(project: str, external_sample_id: str, status: str, batch: str):
+    """Update the status of a sequence given it's corresponding external sample id"""
 
-    seqapi = SequenceApi()
+    samapi = SampleApi()
     sequence_type = SequenceType('genome')
     sample_type = SampleType('blood')
     seq_meta = {'batch': batch}
+    external_sequence_id: Optional[str] = None
 
-    try:
-        sequence_model = SequenceUpdateModel(
-            status=SequenceStatus(status), meta=seq_meta
+    sid_map = samapi.get_sample_id_map_by_internal([external_sample_id])
```

Comment:
This won't work with an external_sample_id. Did you mean 
```suggestion
    sid_map = samapi.get_sample_id_map_by_external(
        project=project, request_body=[external_sample_id]
    )
```
But I think this will also error out at this stage instead of progressing gracefully if the sample doesn't exist. Perhaps a try-except or something?

Reply:
(Reverted update_sample_status per our conversation)